### DISCUSSION TITLE
Is the signIn event guaranteed to be run before we are redirected to our website?

### DISCUSSION THREAD

**Discussion by codinginflow** [nextauthjs/next-auth#7765] [Help]
In my e-commerce website, I want to merge the anonymous shopping cart with the user's shopping cart when they log in. I want this process to finish before the user is redirected to our website to guarantee we fetch the up-to-date cart server-side.

Is the `signIn` event in the `NextAuthOptions` the correct place to do this? Is this callback guaranteed to be run and finished before we are redirected to our own homepage, after a successful social login?

**Comment by tcsidevtools** [comment]
If you mean the signIn callback in ...nextauth route page then yes. That callback controls if the user is actually signed in or not as a final step sort of thing

**Reply by codinginflow** [reply]
I mean the event 
![image](https://github.com/nextauthjs/next-auth/assets/52977034/bba630c2-50c4-4b80-b39b-651f461788af)
